### PR TITLE
Fix retrofit call error

### PR DESCRIPTION
**Pull request by David-Development**
Fix https://github.com/nextcloud/Android-SingleSignOn/issues/280

### CODE REVIEW

**Comment by David-Development** [comment]
@tobiasKaminsky I'm actually not 100% sure whether we should call `onFailure` or return the error response as it is right now. 
I think returning error response is probably better as it is not really a failure when an app is not enabled  - and therefore we are returning an "error" response indicating that the app is not available. 

TLDR; I think it is good as it is right now.

File: src/main/java/com/nextcloud/android/sso/helper/Retrofit2Helper.java [hunk 1]
```diff
@@ -99,6 +101,19 @@ public boolean isCanceled() {
...
+            private Response<T> convertExceptionToResponse(int statusCode, String errorMessage) {
+                ResponseBody body = ResponseBody.create(null, errorMessage);
+                return Response.error(
+                        body,
+                        new okhttp3.Response.Builder()
+                                .body(body)
+                                .code(statusCode)
+                                .message(errorMessage)
+                                .protocol(Protocol.HTTP_1_1)
+                                .request(new Request.Builder().url("http://localhost/" + nextcloudRequest.getUrl()).build())
```
Comment (tobiasKaminsky) [inline src/main/java/com/nextcloud/android/sso/helper/Retrofit2Helper.java:114]: Is this hardcoded localhost correct?

Comment (David-Development) [reply src/main/java/com/nextcloud/android/sso/helper/Retrofit2Helper.java:114]: It's never used anywhere (it's just needed for the error message). I copied this from the reotrofit lib and they are using localhost as well (hardcoded).

**Review by tobiasKaminsky — APPROVED** [review]

**Comment by tobiasKaminsky** [comment]
@matiasdelellis  can you give this a test run?

**Comment by matiasdelellis** [comment]
HI both,
Tomorrow test that.. Thanks.. 😄

**Comment by matiasdelellis** [comment]
Hi both,
Great.. Now get a valid empty response, and I can evaluate the redirect code to know what is happening.

Thanks you so much.. 😄

**Pull request merged**